spec/lib/mobile_application_platform/sign_up/service_spec.rb

```diff
@@ -0,0 +1,258 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'mobile_application_platform/sign_up/service'
+
+describe MobileApplicationPlatform::SignUp::Service do
+  let(:icn) { '10101V964144' }
+  let(:log_prefix) { '[MobileApplicationPlatform][SignUp][Service]' }
+
+  describe '#status' do
+    subject { described_class.new.status(icn:) }
+
+    context 'when an issue occurs with the client request' do
+      let(:expected_error) { Common::Client::Errors::ClientError }
+      let(:expected_error_message) do
+        "#{log_prefix} status failed, client error, status: #{status}, " \
+          "description: #{description}, icn: #{icn}"
+      end
+      let(:status) { 'some-status' }
+      let(:description) { 'some-description' }
+      let(:raised_error) { Common::Client::Errors::ClientError.new(nil, status, { error_description: description }) }
+
+      before do
+        allow_any_instance_of(described_class).to receive(:perform).and_raise(raised_error)
```

Comment:
This might be expedient, but it is a bit of a smell. In general, it is preferred to stub using an instance double rather than stubbing any instance of the class, but the real smell here is that this is stubbing a private method on the system under test.

One approach to remove the smell might be to record and use a specific VCR cassette that will raise the error naturally. Another approach that might work is to do something like what they've done [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/lib/hca/service_spec.rb#L195).

I think either approach would work, but I've never tried the second one before.

Reply:
This is reasonable, I'm going to defer this to a potential future PR to investigate error cases and deal with them more elegantly in code/specs. This service unfortunately can only be tested on the VA network, so it's difficult to test (especially error cases) until we have an interface we can leverage on a dev/staging instance

Reply:
I added a ticket for this